Baseline

obiwankennedy opened the pull request:
Set of changes in order to convert code to modern code and also add mouse input/touchscreen management (could be probably improved).
We can discuss about each change if you want to know why. 
Also I have some auto indentations so, Many lines are changes but in fact it is just the same code.

Pokegali commented:
You may want to check the artifacts, the menus are broken, the button popups as well, and the score is not updating.
Unless it's only for me ?

obiwankennedy commented:
No no it is probably on my side because I chopped in little pieces the code, removed some out of scope parts for the training in order to reduce the complexity, then I rebased on your recent changes and that make a tone of conflicts I solved them but I probably forgot to add what I did removed.
